### ISSUE TITLE
Plugin: grab-canvas

### ISSUE THREAD

**Issue by TimPietrusky** [vcync/modv-3#138]
In modV2 we had the plugin grab-canvas that was used in combination with https://github.com/NERDDISCO/luminave to exchange color-information that is grabbed live from the output-canvas to control lights. 

## Idea: API via GraphQL

Maybe it would make sense to not directly connect to luminave over Websocket, but to provide an API via GraphQL that provides a subscription (which is also websocket), so that also other applications could get that information out of modV.

**Comment by 2xAA** [comment]
grab-canvas is a completely proprietary method to communicate with Luminave.

I think maybe we should look at ways of getting video data out of modV without processing it first, what do you think?

Really like the idea of a GraphQL API.
I'm not sure the GraphQL API would be super necessary for grab-canvas if we were able to use something like a frameserver to share video frames.

Luminave would have to grab the frames and process them separately, but I don't think this would be such a bad thing.

**Comment by TimPietrusky** [comment]
What do you have in mind for a frameserver? Are you thinking about the _Screen Capture API_? 

And don't you think we should process some stuff, so that other applications that don't know how to process the video, but just want to use some processed data, can also benefit from this?

**Comment by 2xAA** [comment]
I didn't have anything in mind for the frameserver exactly. Just putting it out there as a route to explore.

I'm concerned that we may end up inventing a new way of communicating between our audio/visual apps and isolate our apps from playing nicely with other applications.

We may want to process some stuff on modV's side. I'd be more comfortable to standardise the visual output of modV and let applications consume and manipulate however they like. Though I can see the benefit of keeping the existing grab-canvas API in modV.

Were the performance issues of grab-canvas fixed in the end?

Again I really like the idea of GraphQL, I think that could be very useful. Especially if we think about bringing things like a mobile controller to modV again (this was available in versions before 1.5).

**Comment by TimPietrusky** [comment]
@2xAA can we please have a meeting about this and talk about the different scenarios that we want to achieve? And then create the correct tickets.

**Comment by 2xAA** [comment]
Yes, absolutely!

**Comment by TimPietrusky** [comment]
Scale the image down to the correct size and it should scale it down to the exact pixels that we need.

**Comment by 2xAA** [comment]
Yes, this should work nice and fast. No frame server or shader required!

**Comment by TimPietrusky** [comment]
PoC: https://codesandbox.io/s/grab-canvas-image-scale-bcdfr?file=/src/index.js
